### DISCUSSION TITLE
Retreive data separately from `useQuery`

### DISCUSSION THREAD

**Discussion by innovaweb-dev** [urql-graphql/urql#2572] [Q&A] [answered]
Hello urql community

I can't retrieve `customers` and `__type` from `data`.
Thanks,

```js
<script>
import { useQuery } from '@urql/vue';
export default {
	setup() {
		const result = useQuery({
			query: `
                        { customers {lastname email }
                          __type(name : "Customer"){fields {description name }
		}
	}
      `
	});
		console.log(result.data); //OK
		console.log(result.data.customers); //undefined
		console.log(result.data.__type); //undefined
	}
};
</script>
```

![image](https://user-images.githubusercontent.com/2601347/181485673-7640dd5b-4703-43a2-a4c2-e150050bdb7f.png)

**Comment by JoviDeCroock** [comment, marked as answer]
Because it's still a ref here wouldn't you need to do

```js
console.log(result.data.value.customers);
```

When we return a ref like [this](https://github.com/FormidableLabs/urql/blob/main/examples/with-vue3/src/PokemonList.vue#L33) it will unwrap automatically because Vue does this for us but in that setup function you are dealing with the raw ref. TypeScript should be warning you here that customers and __type do not exist 😅 unless there's something wrong with the types and it doesn't think it's a ref there. Internally we use [an unwrap function](https://github.com/FormidableLabs/urql/blob/main/packages/vue-urql/src/utils.ts#L3)

EDIT: yeah should be a ref according to the types as specified [here](https://github.com/FormidableLabs/urql/blob/main/packages/vue-urql/src/useQuery.ts#L41)

**Reply by innovaweb-dev** [reply]
Thanks @JoviDeCroock, I gonna look around Vue.
